trie/utils/nodes.py

```diff
@@ -173,3 +176,43 @@ def encode_leaf_node(value):
     if value is None or value == b'':
         raise ValidationError("Value of leaf node can not be empty")
     return LEAF_TYPE_PREFIX + value
+
+
+def annotate_node(node_body):
```

Comment:
type hints since this is new code?

Reply:
Okay, yeah, I'll look at how carefully I can define the node bodies. I think it's a recursive type definition, which I don't know/remember how to do.